projects/pgai/pgai/vectorizer/loading.py

```diff
@@ -43,13 +44,40 @@ class UriLoading(BaseModel):
     implementation: Literal["uri"]
     column_name: str
     retries: int = 6
+    aws_role_arn: str | None = None
 
     def load(self, row: dict[str, str]) -> LoadedDocument:
         # Note: deferred import to avoid import overhead
+        import boto3
         import smart_open  # type: ignore
 
+        transport_params = None
+        if self.aws_role_arn is not None:
```

Comment:
We need to check that the URI is a valid s3 resource (`s3://*`) and not a local file or any file stored in other service.

Reply:
done

Reply:
done